frontend/server/src/Controllers/QualityNomination.php

```diff
@@ -80,14 +85,33 @@ public static function apiCreate(\OmegaUp\Request $r) {
             throw new \OmegaUp\Exceptions\NotFoundException('problemNotFound');
         }
 
-        if ($r['nomination'] != 'demotion') {
-            // All nominations types, except demotions, are only allowed for
-            // uses who have already solved the problem.
-            if (!\OmegaUp\DAO\Problems::isProblemSolved($problem, (int)$r->identity->identity_id)) {
-                throw new \OmegaUp\Exceptions\PreconditionFailedException('qualityNominationMustHaveSolvedProblem');
+        if ($r['nomination'] !== 'demotion') {
+            if (isset($contents['before_ac']) &&
+                $contents['before_ac'] &&
+                ($r['nomination'] === 'dismissal' ||
+                $r['nomination'] === 'suggestion')
+            ) {
+                // Before AC suggestions or dismissals are only allowed
+                // for users who didn't solve a problem, but tried to.
+                if (\OmegaUp\DAO\Problems::isProblemSolved($problem, (int)$r->identity->identity_id)) {
+                    throw new \OmegaUp\Exceptions\PreconditionFailedException('qualityNominationMustNotHaveSolvedProblem');
```

Comment:
nit:

```suggestion
                    throw new \OmegaUp\Exceptions\PreconditionFailedException(
                        'qualityNominationMustNotHaveSolvedProblem'
                    );
```

igual abajo.